x-pack/plugins/apm/server/routes/service_dashboards/get_services_with_dashboards.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery, termQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { estypes } from '@elastic/elasticsearch';
+import { SERVICE_NAME } from '../../../common/es_fields/apm';
+import {
+  APMEventClient,
+  APMEventESSearchRequest,
+} from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { SavedServiceDashboard } from '../../../common/service_dashboards';
+
+function getSearchRequest(
+  filters: estypes.QueryDslQueryContainer[]
+): APMEventESSearchRequest {
+  return {
+    apm: {
+      events: [ProcessorEvent.metric, ProcessorEvent.transaction],
+    },
+    body: {
+      track_total_hits: false,
+      terminate_after: 1,
+      size: 1,
+      query: {
+        bool: {
+          filter: filters,
+        },
+      },
+    },
+  };
+}
+export async function getServicesWithDashboards({
+  apmEventClient,
+  allLinkedCustomDashboards,
+  serviceName,
+}: {
+  apmEventClient: APMEventClient;
+  allLinkedCustomDashboards: SavedServiceDashboard[];
+  serviceName: string;
+}): Promise<SavedServiceDashboard[]> {
+  const allKueryPerDashboard = allLinkedCustomDashboards.map(({ kuery }) => ({
+    kuery,
+  }));
+  const allSearches = allKueryPerDashboard.map((dashboard) =>
+    getSearchRequest([
+      ...kqlQuery(dashboard.kuery),
```

Comment:
pass the timerange to improve perf

Reply:
fixed [Pass timerange](https://github.com/elastic/kibana/pull/166789/commits/86ae0107d58a003fff5bdbb6b7763b450dddee49)